src/main/java/nl/erwinvaneyk/communication/ConnectorImpl.java

```diff
@@ -34,15 +35,19 @@ public Message sendRequest(Message message, NodeAddress destination) throws Comm
 
     @Override
     public Set<NodeAddress> broadcast(Message message) {
-        Set<NodeAddress> nodes = me.getConnectedNodes();
+        Set<NodeAddress> nodes = new HashSet<>(me.getConnectedNodes());
+        nodes.add(me.getAddress());
 
         return broadcast(message, nodes);
     }
 
     public Set<NodeAddress> broadcast(Message message, String identifierFilter) {
-        Set<NodeAddress> nodes = me.getConnectedNodes()
-                .stream()
-                .filter(n -> n.getIdentifier().matches("(.*)" + identifierFilter + "(.*)")).collect(toSet());
+        Set<NodeAddress> nodes = new HashSet<>(me.getConnectedNodes());
+        nodes.add(me.getAddress());
```

Comment:
No need for creating a new set, as the result `toSet()` of the stream is already provides a new set.